File: DS and Algorithm/AlgoExpert/selectionSort.js
function selectionSort(array) {
  let startIdx = 0;
  while (startIdx < array.length - 1) {
    let smallestIdx = startIdx;
    for (let i = startIdx + 1; i < array.length; i++) {
      if (array[i] < array[smallestIdx]) {
        smallestIdx = i;
      }
    }
    swap(startIdx, smallestIdx, array);
    startIdx++;
  }
  return array;
}

function swap(i, j, array) {
  let temp = array[j];
  array[j] = array[i];
  array[i] = temp;
}

console.log(selectionSort([10, 8, 5, 2, 9, 5, 6, 3]));

File: AlgoExpert/Easy/selectionSort.py
def selectionSort(array):
    currentIdx = 0
    while(currentIdx < len(array) - 1):
        smallestIdx = currentIdx
        for i in range(currentIdx+1, len(array)):
            if array[smallestIdx] > array[i]:
                smallestIdx = i
        swap(currentIdx, smallestIdx, array)
        currentIdx += 1
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]


print(selectionSort([1, 3, 1, 3, 2, 5]))
